### DISCUSSION TITLE
txt.py里面的ssd_voc_5000_plus.pth文件

### DISCUSSION THREAD

**Discussion by jdbsc** [PengfeiM/Fatigue-Driven-Detection-Based-on-CNN#13] [Q&A]
这个文件是手动添加吗 要用哪个测试数据？

**Comment by PengfeiM** [comment]
txt.py?

**Comment by jdbsc** [comment]
sorry 不是txt 是test.py

**Comment by PengfeiM** [comment]
这个应该是训练好的权重文件

**Comment by jdbsc** [comment]
可是train.py 训练生成的文件格式为 ssd30_VOC_1000（迭代次数）.pth  没有ssd_voc_5000_plus.pth这种形式的

**Comment by jdbsc** [comment]
另外数据集中共有4000多张图片   为何eval.py输出时显示总共1283张
@PengfeiM

**Comment by PengfeiM** [comment]
1. **权重文件问题**
那可能是我改过代码，然后那一块后来没用就没改了。
2. **数据集问题**
因为4000多张图片一部分用来做训练集，一部分用来做测试集。    训练和测试的图片是随机选的，在数据集里面有个txt.py用来生成训练测试列表。自己可以调整比例。

**Reply by Guojc-source** [reply]
你好，我想请问一下问题1 的没用是什么意思，是不需要detection也可以跑代码吗

**Reply by PengfeiM** [reply]
倒不是detection没用。
应该是指哪个单个图片检测的代码里面，ssd_.......5000_plus.pth哪个文件是我学习CNN的时候训练出来的一个文件。
后来做视频检测的时候就没再用图片检测的代码了，所以没改。

**Comment by jdbsc** [comment]
好的

**Comment by jdbsc** [comment]
作者 我想问一下你CNN用的哪个模型

**Comment by PengfeiM** [comment]
> 作者 我想问一下你CNN用的哪个模型

ssd，

**Comment by jdbsc** [comment]
我想问一下这个CNN、模型相比传统CNN模型的改进点在哪

**Comment by Guojc-source** [comment]
大佬你好，我试过用文件夹中的ssd_5000来替代这个5000_plus但是直接报错mismatch了，如果不得不用个权重文件去替代的话请问有什么思路呢




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "PengfeiM/Fatigue-Driven-Detection-Based-on-CNN"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年4月7日(星期四) 上午10:36
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [PengfeiM/Fatigue-Driven-Detection-Based-on-CNN] txt.py里面的ssd_voc_5000_plus.pth文件 (Discussion #13)





 
倒不是detection没用。
 应该是指哪个单个图片检测的代码里面，ssd_.......5000_plus.pth哪个文件是我学习CNN的时候训练出来的一个文件。
 后来做视频检测的时候就没再用图片检测的代码了，所以没改。

**Comment by PengfeiM** [comment]
你看下github，我把那个文件找到了，补上了。

发件人: Guojc-source ***@***.***>
发送时间: 2022年4月7日 15:53
收件人: PengfeiM/Fatigue-Driven-Detection-Based-on-CNN ***@***.***>
抄送: Ma Pengfei ***@***.***>; Mention ***@***.***>
主题: Re: [PengfeiM/Fatigue-Driven-Detection-Based-on-CNN] txt.py里面的ssd_voc_5000_plus.pth文件 (Discussion #13)

大佬你好，我试过用文件夹中的ssd_5000来替代这个5000_plus但是直接报错mismatch了，如果不得不用个权重文件去替代的话请问有什么思路呢




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "PengfeiM/Fatigue-Driven-Detection-Based-on-CNN" ***@***.***&gt<mailto:***@***.***&gt>;;
发送时间:&nbsp;2022年4月7日(星期四) 上午10:36
***@***.***&gt<mailto:***@***.***&gt>;;
***@***.******@***.***&gt<mailto:***@***.******@***.***&gt>;;
主题:&nbsp;Re: [PengfeiM/Fatigue-Driven-Detection-Based-on-CNN] txt.py里面的ssd_voc_5000_plus.pth文件 (Discussion #13)






倒不是detection没用。
应该是指哪个单个图片检测的代码里面，ssd_.......5000_plus.pth哪个文件是我学习CNN的时候训练出来的一个文件。
后来做视频检测的时候就没再用图片检测的代码了，所以没改。

**Reply by NightShy** [reply]
> 多谢大佬！！！！！有了权重文件后已经可以成功运行

可不可以教一下小白拿到这个程序该如何运行啊，非常感谢

**Comment by Guojc-source** [comment]
好的，我这就去试试，万分感谢大佬！！！




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "PengfeiM/Fatigue-Driven-Detection-Based-on-CNN"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年4月7日(星期四) 下午3:54
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [PengfeiM/Fatigue-Driven-Detection-Based-on-CNN] txt.py里面的ssd_voc_5000_plus.pth文件 (Discussion #13)





 你看下github，我把那个文件找到了，补上了。 
 
 发件人: Guojc-source ***@***.***&gt; 
 发送时间: 2022年4月7日 15:53 
 收件人: PengfeiM/Fatigue-Driven-Detection-Based-on-CNN ***@***.***&gt; 
 抄送: Ma Pengfei ***@***.***&gt;; Mention ***@***.***&gt; 
 主题: Re: [PengfeiM/Fatigue-Driven-Detection-Based-on-CNN] txt.py里面的ssd_voc_5000_plus.pth文件 (Discussion #13) 
 
 大佬你好，我试过用文件夹中的ssd_5000来替代这个5000_plus但是直接报错mismatch了，如果不得不用个权重文件去替代的话请问有什么思路呢 
 
 
 
 
 ------------------&amp;nbsp;原始邮件&amp;nbsp;------------------ 
 发件人: "PengfeiM/Fatigue-Driven-Detection-Based-on-CNN" ***@***.***&amp;gt<mailto:***@***.***&amp;gt&gt;;; 
 发送时间:&amp;nbsp;2022年4月7日(星期四) 上午10:36 
 ***@***.***&amp;gt<mailto:***@***.***&amp;gt&gt;;; 
 ***@***.******@***.***&amp;gt<mailto:***@***.******@***.***&amp;gt&gt;;; 
 主题:&amp;nbsp;Re: [PengfeiM/Fatigue-Driven-Detection-Based-on-CNN] txt.py里面的ssd_voc_5000_plus.pth文件 (Discussion #13) 
 
 
 
 
 
 
 倒不是detection没用。 
 应该是指哪个单个图片检测的代码里面，ssd_.......5000_plus.pth哪个文件是我学习CNN的时候训练出来的一个文件。 
 后来做视频检测的时候就没再用图片检测的代码了，所以没改。

**Comment by yun189** [comment]
请问ssd检测到人脸之后，怎么算出perclos呢？这里没看懂
